app/src/main/java/intellistart/interviewplanning/model/booking/validation/BookingValidator.kt

```diff
@@ -48,41 +50,52 @@ class BookingValidator(
         }
     }
 
-    private fun checkBookingForOverlapsWithSlots(booking: Booking) {
-        val periodOfInterviewer: Period = slotService.getById(booking.interviewerSlotId).period
-        val periodOfCandidate: Period = slotService.getById(booking.candidateSlotId).period
-
+    private fun checkBookingForOverlapsWithSlots(booking: Booking): Mono<Unit> = Mono.zip(
+        slotService.getById(booking.interviewerSlotId),
+        slotService.getById(booking.candidateSlotId)
+    ).flatMap { tuple ->
```

Comment:
**NIT**
```suggestion
    ).handle{ (periodOfInterviewer, periodOfCandidate), sink ->
        // ...
        if (...) {
            sink.error(SomeException(...))
        } else {
            sink.next(Unit)
        }
```

`handle` has better performance rather than `flatMap`

Optionally, you can watch
https://www.youtube.com/watch?v=R8bY3i_ptM0&ab_channel=JPoint%2CJoker%D0%B8JUGru